gitlab/datadog_checks/gitlab/gitlab.py

```diff
@@ -63,6 +63,9 @@ def check(self, instance):
         for check_type in self.ALLOWED_SERVICE_CHECKS:
             self._check_health_endpoint(instance, check_type, custom_tags)
 
+        if self.is_metadata_collection_enabled():
```

Comment:
nit: you can move the if inside submit_version so the method is noop if disabled

Reply:
Sure!